test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/TempDataApplicationModelProviderTest.cs

```diff
@@ -0,0 +1,125 @@
+
+            // Assert
+            Assert.NotNull(saveTempDataPropertyFilterFactory);
+            var tempDataPropertyHelper = Assert.Single(saveTempDataPropertyFilterFactory.TempDataProperties);
+            Assert.Equal("Test2", tempDataPropertyHelper.Name);
+            Assert.NotNull(tempDataPropertyHelper.Property.GetCustomAttribute<TempDataAttribute>());
+        }
+
+        [Fact]
+        public void ThrowsInvalidOperationException_PrivateSetter()
+        {
+            // Arrange
+            var provider = new TempDataApplicationModelProvider();
+            var defaultProvider = new DefaultApplicationModelProvider(new TestOptionsManager<MvcOptions>());
+
+            var context = new ApplicationModelProviderContext(new[] { typeof(TestController_PrivateSet).GetTypeInfo() });
+            defaultProvider.OnProvidersExecuting(context);
+
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(() =>
+            {
+                provider.OnProvidersExecuting(context);
+            });
+
+            Assert.Equal($"The '{typeof(TestController_PrivateSet).FullName}.{nameof(TestController_NonPrimitiveType.Test)}' property with TempDataAttribute is invalid. A property using TempDataAttribute must have a public getter and setter.", exception.Message);
+        }
+
+        [Fact]
+        public void ThrowsInvalidOperationException_NonPrimitiveType()
+        {
+            // Arrange
+            var provider = new TempDataApplicationModelProvider();
+            var defaultProvider = new DefaultApplicationModelProvider(new TestOptionsManager<MvcOptions>());
+
+            var context = new ApplicationModelProviderContext(new[] { typeof(TestController_NonPrimitiveType).GetTypeInfo() });
+            defaultProvider.OnProvidersExecuting(context);
+
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(() =>
+            {
+                provider.OnProvidersExecuting(context);
+            });
+
+            Assert.Equal($"The '{typeof(TestController_NonPrimitiveType).FullName}.{nameof(TestController_NonPrimitiveType.Test)}' property with TempDataAttribute is invalid. A property using TempDataAttribute must be of primitive or string type.", exception.Message);
+        }
+
+        public class TestController_OneTempDataProperty
+        {
+            public string Test { get; set; }
+
+            [TempData]
+            public string Test2 { get; set; }
+        }
+
+        public class TestController_TwoTempDataProperties
+        {
+            [TempData]
+            public string Test { get; set; }
+
+            [TempData]
+            public string Test2 { get; set; }
+        }
+
+        public class TestController_PrivateSet : Controller
+        {
+            [TempData]
+            public string Test { get; private set; }
+        }
+
+        public class TestController_NonPrimitiveType : Controller
```

Comment:
Could you add a scenario where some of the properties are valid and the others aren't allowed?